FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/DicomApiDataClient.cs

```diff
@@ -4,115 +4,177 @@
+                    throw new FhirSearchException("Get server access token failed", ex);
+                }
             }
+
+            return await GetResponseFromHttpRequestAsync(searchUri, accessToken, cancellationToken);
         }
 
-        private async Task<List<ChangeFeedEntry>> GetChangeFeedEntriesAsync(ChangeFeedOptions changeFeedOptions, CancellationToken cancellationToken = default)
+        private Uri CreateSearchUri(BaseDicomApiOptions dicomApiOptions)
         {
-            if (changeFeedOptions.IncludeMetadata)
+            string serverUrl = _dataSource.FhirServerUrl;
+
+            var baseUri = new Uri(serverUrl);
+
+            var uri = new Uri(baseUri, dicomApiOptions.RelativeUri());
+
+            // the query parameters is null for metadata
+            if (dicomApiOptions.QueryParameters == null)
             {
-                _diagnosticLogger.LogError("includemetadata should be set to false in changefeed.");
-                _logger.LogInformation("includemetadata should be set to false in changefeed.");
-                throw new FhirSearchException("includemetadata should be set to false in changefeed.");
+                return uri;
             }
 
+            return uri.AddQueryString(dicomApiOptions.QueryParameters);
+        }
+
+        private async Task<string> GetResponseFromHttpRequestAsync(Uri uri, string accessToken = null, CancellationToken cancellationToken = default)
+        {
             try
             {
-                var response = await _httpClient.GetAsync(
-                    $"/changefeed?offset={changeFeedOptions.Offset}&limit={changeFeedOptions.Limit}&includeMetadata={changeFeedOptions.IncludeMetadata}",
-                    cancellationToken);
+                var searchRequest = new HttpRequestMessage(HttpMethod.Get, uri);
+                if (accessToken != null)
+                {
+                    // Currently we support accessing service endpoints with Managed Identity.
+
+                    // The thread-safe AzureServiceTokenProvider class caches the token in memory and retrieves it from Azure AD just before expiration.
+                    // https://docs.microsoft.com/en-us/dotnet/api/overview/azure/service-to-service-authentication#using-the-library
+                    searchRequest.Headers.Authorization = new AuthenticationHeaderValue("Bearer", accessToken);
+                }
+
+                HttpResponseMessage response = await _httpClient.SendAsync(searchRequest, cancellationToken);
+                response.EnsureSuccessStatusCode();
+
+                _logger.LogInformation("Successfully retrieved result for url: '{url}'.", uri);
+
+                return await response.Content.ReadAsStringAsync(cancellationToken);
+            }
+            catch (HttpRequestException hrEx)
+            {
+                switch (hrEx.StatusCode)
+                {
+                    case HttpStatusCode.Unauthorized:
```

Comment:
We can also add .Forbidden to handle '403' code which is a missing part in the old FHIR api handler

Reply:
Pre offline discussion, will fix this in later PR